File: _CodeTopics/LeetCode_contest/biweekly/35/35_1.py
class Solution(object):
    def sumOddLengthSubarrays(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        
        length = len(arr)
        res = 0
        for currlen in range(1, length+1, 2):
            for i in range(length-currlen+1):
                for k in range(i, i+currlen):
                    res += arr[k]
        return res
    
"""
https://leetcode-cn.com/submissions/detail/109623827/

96 / 96 个通过测试用例
状态：通过
执行用时: 100 ms
内存消耗: 12.3 MB
"""

File: _CodeTopics/LeetCode/1401-1600/001588--biweekly35_1/tran/35_1.cpp
class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {

        // 第 35 场双周赛第一题： https://leetcode-cn.com/submissions/detail/109623827/

        int length = arr.size();
        int res = 0;
        for (int currlen = 1; currlen < length + 1; currlen += 2) {
            for (int i = 0; i < length - currlen + 1; ++i) {
                for (int k = i; k < i + currlen; ++k) {
                    res += arr[k];
                }
            }
        }
        return res;
    }
};

/*
https://leetcode-cn.com/submissions/detail/212713159/

96 / 96 个通过测试用例
状态：通过
执行用时: 4 ms
内存消耗: 8 MB

执行用时：4 ms, 在所有 C++ 提交中击败了76.28%的用户
内存消耗：8 MB, 在所有 C++ 提交中击败了87.21%的用户
*/
